src/sentry/middleware/customer_domain.py

```diff
@@ -52,16 +52,25 @@ def _resolve_redirect_url(request, activeorg):
     if redirect_subdomain:
         redirect_url = generate_organization_url(activeorg)
     result = resolve(request.path)
-    org_slug_path_mismatch = (
-        result.kwargs
-        and "organization_slug" in result.kwargs
-        and result.kwargs["organization_slug"] != activeorg
+    org_slug_path_mismatch = result.kwargs and (
```

Comment:
To make this easier to understand, with this solution, if ID is passed in `organization_id_or_slug` and it is not the correct ID of the active org redirect_url stays activeorg url but `kwargs["organization_id_or_slug"]` remain the wrong ID so this leads to 404. I think it's expected but just clarifying.

Reply:
Yep thats correct. In general, this scenario shouldn't be happening because the org_id should only be used in subdomain-less queries.